### PR TITLE
Check navigator.ml is defined before trying to run WebNN tests

### PR DESCRIPTION
**Pull request by jgraham**
Otherwise these tests end up taking multiple hours in browsers that don't support.

### CODE REVIEW

**Review by foolip — APPROVED** [review]

**Review by Ms2ger — APPROVED** [review]
Ok, but isn't this something for the "precondition failed" thing we added a while ago?

**Review by gobig1983 — APPROVED** [review]

**Review by Honry — APPROVED** [review]
👍

**Comment by BruceDai** [comment]
@jgraham Thanks a lot for this fixing!

File: webnn/resources/utils.js [hunk 1]
```diff
@@ -825,6 +825,10 @@ const run = async (operationName, context, builder, resources, buildFunc) => {
  * @param {String} deviceType - The execution device type for this test
  */
 const testWebNNOperation = (operationName, buildFunc, deviceType = 'cpu') => {
+  test(() => assert_not_equals(navigator.ml, undefined, "ml property is defined on navigator"));
```
Comment (BruceDai) [inline webnn/resources/utils.js:828]: I'd like to specially add this **ml** property checking test as a validate test under validation_tests folder, not on here for all invoking this testWebNNOperation function for each operation test.

**Comment by jgraham** [comment]
So I merged this, since it's causing problems in the short term.

In response to the comments:
* I thought we decided `PRECONDITION_FAILED` was for actually optional things? If you implement webNN presumably these tests are for mandatory features, so I think not having the API is a normal kind of failure on that basis.
* The reason I added this to all the tests is because we need to short-circuit in the failure case (otherwise we were taking multiple hours to fail all the tests). We also need some indication to users why the test failed (i.e. just not running any tests isn't really OK). I'm happy with other solutions as long as they preserve the runtime in non-implementing browsers.

**Pull request merged**